src/jdk.compiler/share/classes/com/sun/tools/javac/code/Type.java

```diff
@@ -2408,27 +2408,22 @@ public <R, P> R accept(TypeVisitor<R, P> v, P p) {
         }
     }
 
-    public static class UnknownType extends Type {
+    public static class UnknownType extends ErrorType {
 
-        public UnknownType() {
+        public UnknownType(ClassSymbol unknownSymbol) {
             // Unknown is a synthesized internal type, so it cannot be
             // annotated.
-            super(null, List.nil());
-        }
-
-        @Override
-        public TypeTag getTag() {
-            return UNKNOWN;
+            super(unknownSymbol, noType);
         }
 
         @Override @DefinedBy(Api.LANGUAGE_MODEL)
         public <R, P> R accept(TypeVisitor<R, P> v, P p) {
             return v.visitUnknown(this, p);
         }
 
-        @Override
-        public boolean isPartial() {
-            return true;
+        @Override @DefinedBy(Api.LANGUAGE_MODEL)
```

Comment:
Modelling-wise, I'm not sure. It almost seems as if `UnknownType` should be subclassed by `ErrorType` and not the other way around. Is there a specific reason you went for this?

Reply:
I would see the `UnknownType` as a specific kind of an erroneous type (and hence `UnknownType` being a subtype or erroneous, as it is here).

But, even from a practical point of view - `ErrorType` basically must extend `ClassType`, I think, so implanting `UnknownType` in there seems difficult.

Reply:
I guess I'm a bit surprised by the typeKind - which is OTHER (you added the override to keep compatibility with what was there before). In that sense, it doesn't seem like `UnknownType` "is a" `ErrorType`, because its tag seems "less general" than the one in `ErrorType`.